source/exe/main.cc

```diff
@@ -33,10 +35,32 @@ class ProdComponentFactory : public ComponentFactory {
 
 } // Server
 
+int validateConfigAndExit(OptionsImpl& options, Server::ProdComponentFactory& component_factory,
+                          const LocalInfo::LocalInfoImpl& local_info) {
+  Server::ValidationHotRestart restarter;
+  Logger::Registry::initialize(options.logLevel(), restarter.logLock());
+  Stats::HeapRawStatDataAllocator alloc;
+  Stats::ThreadLocalStoreImpl stats_store(alloc);
+
+  Server::ValidationInstance server(options, restarter, stats_store, restarter.accessLogLock(),
+                                    component_factory, local_info);
+  std::cerr << "configuration '" << options.configPath() << "' OK" << std::endl;
```

Comment:
Done. (My original thinking was to have success and failure outputs go the same place, and this is now the only reference to stdout in non-test code -- but that's probably fine.)